addons/website/tests/test_page.py

```diff
@@ -235,3 +239,19 @@ def test_search(self):
             [{'loc': p['loc']} for p in pages],
             [{'loc': '/page_1'}]
         )
+
+    def test_homepage_not_slash_url(self):
+        website = self.env['website'].browse([1])
+        # Set another page (/page_1) as homepage
+        website.write({
+            'homepage_id': self.page.id,
+            'domain': f"http://{HOST}:{config['http_port']}",
+        })
+        assert self.page.url != '/'
+
+        r = self.url_open('/')
+        self.assertEqual(r.status_code, 200,
+                         "There should be no crash when a public user is accessing `/` which is rerouting to another page with a different URL.")
```

Comment:
So that we have the response body in the logs in case of error

```suggestion
        r = self.url_open('/')
        r.raise_for_status()
        self.assertEqual(r.status_code, 200,
                         "There should be no crash when a public user is accessing `/` which is rerouting to another page with a different URL.")
```